Robust.Client/Graphics/Clyde/Clyde.cs

```diff
@@ -80,6 +85,69 @@ public override bool Initialize()
             return true;
         }
 
+        public void ReloadShaders()
+        {
+            if (IoCManager.Resolve<IClientConsole>() is IDebugConsole con)
+            {
+                con.AddLine("Reloading base shaders and reinitializing lighting...");
+            }
+            InitLightingAndShaders();
+        }
+
+        private bool _watchingShaders;
+
+        private FileSystemWatcher _shaderWatcher;
+
+        private bool _reloadShadersQueued;
+
+        public void WatchShadersAndReload(bool enabled)
+        {
+            if (enabled == _watchingShaders)
+            {
+                return;
+            }
+
+            if (enabled)
+            {
+                var path = GetInternalShaderPath("");
+                _shaderWatcher = new FileSystemWatcher(path);
+                _shaderWatcher.Filters.Add("*.vert");
+                _shaderWatcher.Filters.Add("*.frag");
+                _shaderWatcher.Filters.Add("*.swsl");
+                _shaderWatcher.Changed += (_, args) =>
+                {
+                    if (_reloadShadersQueued)
+                    {
+                        return;
+                    }
+
+                    var name = args.Name;
+
+                    if (!name.EndsWith(".vert")
+                        && !name.EndsWith(".frag"))
+                    {
+                        return;
+                    }
+
+                    _reloadShadersQueued = true;
+                    SynchronizationContext.Current.Post(o =>
+                    {
+                        ((Clyde) o).ReloadShaders();
```

Comment:
```suggestion
                        ReloadShaders();
```

Just use closure capture man. If you wanna cut out the allocation I feel the need to point out the next line is using it anyways so passing `this` as state object is just harder to read.